### ISSUE TITLE
Fine-tuning a pruned model

### ISSUE THREAD

**Issue by Megh-Thakkar** [huggingface/transformers#29227]
### Feature request

I used [OWL](https://github.com/luuyin/OWL) to prune a Mistral-7b model and would like to further train this pruned model. Is there a way to pass a global mask (based on the pruned model) to ensure those weights don't get updated, particularly when using DDP.

**Note**:  We cannot update the parameter list of the optimizer as the pruning is not structured.

### Motivation

Pruning helps with creating more efficient models, and it would be interesting to analyze the effects of further training pruned models.

### Your contribution

We would be glad to submit a PR for this use case.

**Comment by ArthurZucker** [comment]
FYI @amyeroberts and @NielsRogge

**Comment by amyeroberts** [comment]
@ArthurZucker I think the OWL being referred to here is a library for pruning, rather than owl-vit 🦉 

@Megh-Thakkar I don't know - maybe @younesbelkada has an idea?

**Comment by younesbelkada** [comment]
Hi @Megh-Thakkar 
Thanks for the feature request! 
Currently indeed this is not supported out of the box, also I am not really familiar with OWL library. Would you like to sketch up a draft API of what you have in mind in order to support training pruned models? Can you also elaborate more on

> Note: We cannot update the parameter list of the optimizer as the pruning is not structured.
